osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/ReverseArrowPiece.cs

```diff
@@ -37,7 +39,10 @@ public ReverseArrowPiece()
             };
         }
 
-        protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint, EffectControlPoint effectPoint, TrackAmplitudes amplitudes) =>
-            Child.ScaleTo(1.3f).ScaleTo(1f, timingPoint.BeatLength, Easing.Out);
+        protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint, EffectControlPoint effectPoint, TrackAmplitudes amplitudes)
+        {
+            if (ShouldFollowBeats)
```

Comment:
i think you may be able to resolve the `DrawableHitObject` and check its hit state rather than require this manual bool flow. Should feel a bit more logical.

Reply:
Oh, didn't know the hit object is injected. Will try.

Reply:
yep, a bit hidden:

https://github.com/ppy/osu/blob/9477629a5686e33881bf0c0d28ad4f0c93bda4a4/osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs#L24